overlord/snapstate/snapmgr.go

```diff
@@ -510,6 +521,8 @@ func (m *SnapManager) ensureRefreshes() error {
 	}
 	chg.Set("snap-names", updated)
 	chg.Set("api-data", map[string]interface{}{"snap-names": updated})
+
+	m.nextRefresh = nil
```

Comment:
Shouldn't we ensure this is stopped before we unset it?  Otherwise the function is still called.